compositor_common/src/error.rs

```diff
@@ -44,6 +44,58 @@ impl Display for UnusedNodesError {
     }
 }
 
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum ConstraintsValidationError {
```

Comment:
The name is a bit misleading when this error is happening it means that constraints were not satisfied. It's fare to say that it's a validation error, but if that is the cases then it's scene validation error.

You could name it e.g. UnsatisfiedConstraintsError